fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneConcurrency.java

```diff
@@ -119,7 +119,7 @@ private LuceneConcurrency() {
     /**
      * An exception that is thrown when the async to sync operation times out.
      */
-    public static class AsyncToSyncTimeoutException extends RecordCoreException {
+    public static class AsyncToSyncTimeoutException extends RecordCoreTimeoutException {
```

Comment:
Other than backwards compatibility, is there any reason to keep this class around? Should we mark it as `API.Status.DEPRECATED`, and then, in the future have lucene throw `RecordCoreTimeoutException` instead?